### DISCUSSION TITLE
🧐 Incorrect type definition for "hooks" callback in v16 index.d.ts?

### DISCUSSION THREAD

**Discussion by douglascayers** [octokit/octokit.js#1769] [General] [answered]
We are still using v16.43.1 of this incredible library and for "insert reasons here" we aren't able to prioritize upgrading to v17 or later version. With that in mind, I've come across an issue using **hook.wrap('request', callback)** method on `Octokit` client defined [here](https://raw.githubusercontent.com/octokit/rest.js/v16.43.1/index.d.ts).

At line 33,707 the return type for the callback passed to `wrap` is `void` as shown below:

![image](https://user-images.githubusercontent.com/4142577/84840957-9ee01780-b006-11ea-9e51-4f394b2afdbd.png)

But examples [here](https://octokit.github.io/rest.js/v16#hooks) and [here](https://github.com/octokit/auth-oauth-app.js/blob/master/src/hook.ts) show that the callback should return something.

In typescript, it or my linter, is complaining that I can't return something from a method expected to be `void`.

I think line 33,707 should have `Promise<Octokit.Response<any>>` return type since it'd be returning what the `request` function argument returns, correct?

![image](https://user-images.githubusercontent.com/4142577/84841209-4cebc180-b007-11ea-88f7-33c4670778e4.png)

If that's the case, would you be able to release a minor or patch release of the v16 line? Since index.d.ts is generated, I think the _actual_ code change would occur in `scripts/update-endpoints/templates/index.d.ts.tpl` at [line 323](https://github.com/octokit/rest.js/blob/v16.43.1/scripts/update-endpoints/templates/index.d.ts.tpl#L323).

Thanks!

**Comment by gr2m** [comment]
Thanks Doug for reporting the issue. The `"wrap"` hook needs to indeed return something, while the others don't have to, I'll look into it

**Comment by gr2m** [comment, marked as answer]
See my pull request at https://github.com/octokit/rest.js/pull/1770
